### ISSUE TITLE
Kiribati vax: DOB field of Patient edit is showing invalid if we let dob update with age field

### ISSUE THREAD

**Issue by nisha-dangol** [openmsupply/mobile#4892]
## Describe the bug
It seems that while entering patient details for a new patient record, using mobile keyboard "Next" button (-> arrow button in keyboard), it won't allow to proceed to next page even when all required fields are already entered.

### To reproduce
Steps to reproduce the behaviour:
1. Go to Vaccinations>>New patient
2. Add all required fields and make sure you use Next button (-> arrow button of mobile keyboard) when filling up the details.
3. After all required fields are entered, click `Next` button on bottom-left of the app.
4. See error: see toast message about required fields not completed.
5. Go back to date fields and re-enter values, this time don't click on any Next button(-> arrow button of mobile keyboard) and click `Next` button again. This time it works! Weird?

### Expected behaviour
Don't go weird like that?

### Proposed Solution
Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info
- App version: latest

**Comment by chastichno** [comment]
Can't really reproduce it from the virtual device - will try tomorrow on the real tablet

**Comment by chastichno** [comment]
@nisha-dangol I still cannot replicate it, can you please attach a video recording?

**Comment by sworup** [comment]
@nisha-dangol @chastichno This issue is not related to next button. I have tested it. What is happening is you are setting the age of the user, say 23. The dob field is auto updated, but at the end when we try to submit the form the dateOfBirth that was calculated from age field somehow could not be read properly by the validator and is showing as invalid. 

I used the keyboard next button to pass through all of the field and check which fields are shown as invalid and it is dob only. So we have fix this bug there.
